tests/transactions/publish-auth-capability.ts

```diff
@@ -0,0 +1,69 @@
+import {
+    TransactionFunctionParams,
+    sendTransaction,
+    FlowConstants,
+  } from '../../utils/flow'
+  
+  const CODE = (address: string) => `\
+    #allowAccountLinking
+    import AxelarGovernanceService from ${address}
+
+    transaction(recipient: Address) {
+    
+    let accountCap: Capability<&AuthAccount>
+        prepare(signer: AuthAccount) {
+            let hostPrivatePath = PrivatePath(identifier: "HostAccount_".concat(recipient.toString()))!
+
+            if !signer.getCapability<&AuthAccount>(AxelarGovernanceService.UpdaterContractAccountPrivatePath).check() {
+                signer.unlink(AxelarGovernanceService.UpdaterContractAccountPrivatePath)
+                self.accountCap = signer.linkAccount(AxelarGovernanceService.UpdaterContractAccountPrivatePath)
+                    ?? panic("Problem linking AuthAccount Capability")
+            } else {
+                self.accountCap = signer.getCapability<&AuthAccount>(AxelarGovernanceService.UpdaterContractAccountPrivatePath)
+            }
+            
+            assert(self.accountCap.check(), message: "Invalid AuthAccount Capability retrieved")
+
+            if signer.type(at: AxelarGovernanceService.HostStoragePath) == nil {
+              signer.save(
+                  <- AxelarGovernanceService.createNewHost(accountCap: self.accountCap),
+                  to: AxelarGovernanceService.HostStoragePath
+              )
+            }
+            if !signer.getCapability<&AxelarGovernanceService.Host>(hostPrivatePath).check() {
+                signer.unlink(hostPrivatePath)
+                signer.link<&AxelarGovernanceService.Host>(hostPrivatePath, target: AxelarGovernanceService.HostStoragePath)
+            }
+            let hostCap = signer.getCapability<&AxelarGovernanceService.Host>(hostPrivatePath)
+    
+            assert(hostCap.check(), message: "Invalid Host Capability retrieved")
+    
```

Comment:
Covering case where signer has a Host configured for another account. Unlikely, but it's the only place we have enough visibility to check for it.

```suggestion
    assert(hostCap.borrow()!.getHostAddress()! == signer.address, message: "Host is configured for unexpected account")
```